plugins/vpp/ifplugin/data_resync.go

```diff
@@ -232,8 +232,8 @@ func (plugin *BFDConfigurator) ResyncSession(nbSessions []*bfd.SingleHopBFD_Sess
 	plugin.clearMapping()
 
 	// Dump all BFD vppSessions
-	vppSessions, err := plugin.DumpBfdSessions()
-	if err != nil {
+	vppBfdSessions, err := plugin.bfdHandler.DumpBfdSessions()
+	if err != nil || vppBfdSessions == nil {
```

Comment:
Isn't this going to skip setting BFD from northbound if `DumpBfdSessions` returns `nil`?

Reply:
no, the reason is vppBfdSessions is never nil unless it returns an error. That means the check is useless, I'll remove it